packages/navigator/src/electron-browser/electron-navigator-menu-contribution.ts

```diff
@@ -54,6 +64,28 @@ export class ElectronNavigatorMenuContribution implements MenuContribution, Comm
             isEnabled: uri => !!this.workspaceService.getWorkspaceRootUri(uri),
             isVisible: uri => !!this.workspaceService.getWorkspaceRootUri(uri),
         }));
+        commands.registerCommand(OPEN_WITH_SYSTEM_APP, UriAwareCommandHandler.MonoSelect(this.selectionService, {
+            execute: async uri => {
+                this.openWithSystemApplication(uri);
+            },
+            isEnabled: uri => !!this.workspaceService.getWorkspaceRootUri(uri),
```

Comment:
Why do we care whether the file is in the workspace? As long as it's a file uri, why not try to open it?

Reply:
I thought, because right now it is only possible to call "Open with..." from the navigator it would make sense to restrict the command to WS files. But yes, you are right, why not to allow for all URIs? I will change it.